src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIProxy.cs

```diff
@@ -240,11 +240,11 @@ public uint WritePacket(SNIHandle handle, SNIPacket packet, bool sync)
         {
             if (sync)
             {
-                return handle.Send(packet.Clone());
+                return handle.Send(packet);
```

Comment:
This doesn't seem right. Earlier, we were setting the _outBuff into the SNIPacket and then sending a Clone of the SNI packet over the network. Now we are sending the reference to _outBuff over the wire.
This can cause issues if _outBuff in TdsParserStateObject is changed and the data transmission is still in progress.

Reply:
> if _outBuff in TdsParserStateObject is changed and the data transmission is still in progress.

Can that situation actually happen?
If so, `handle.Send(packet.Clone());` should be atomic.

Reply:
Why should it be atomic? There is a buffer that State Object operates on, and when that buffer is sent to network, it should be dissociated, since the state object can continue to modify it. I am not talking about multiple threads calling this method, but data modifications at multiple locations.